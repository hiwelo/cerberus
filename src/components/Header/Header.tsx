import React, { FunctionComponent } from 'react';
import { Heading } from '..';
import { HeaderElement } from './components';

export const Header: FunctionComponent = ({ children }) => (
  <HeaderElement>
    <Heading>{children}</Heading>
  </HeaderElement>
);
