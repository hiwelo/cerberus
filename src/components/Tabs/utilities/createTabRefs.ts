import { useRef } from 'react';
import { TabsProps } from '../types';

/**
 * Creates a series of React Ref for each panel part of the Tabs defined for
 * this `Tabs` component
 *
 * @param tabs The dictionary of Tabs for the related Tabs component
 */
export function createTabRefs(tabs: TabsProps['tabs']): TabsProps['tabs'] {
  // Transforms the TabListDictionary as an array of tabs
  const tabsEntries = Object.entries(tabs);
  const tabsWithRefs: TabsProps['tabs'] = {};

  // Loops over each tab and creates a React Ref for each panel HTMLElement
  tabsEntries.forEach(([tabKey, tabInfo]) => {
    const refPanel = useRef<HTMLDivElement>(null);

    tabsWithRefs[tabKey] = { ...tabInfo, refPanel };
  });

  return tabsWithRefs;
}
