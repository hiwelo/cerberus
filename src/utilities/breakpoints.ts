/** List of breakpoints in use in our styled components */
export const breakpoints = {
  xs: '0px',
  md: '540px',
};
