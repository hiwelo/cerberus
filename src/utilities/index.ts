export { breakpoints } from './breakpoints';
export { listNodesFromConnection } from './listNodesFromConnection';
