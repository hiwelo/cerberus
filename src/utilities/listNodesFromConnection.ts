/** Describes the generic structure of a GraphQL edge */
type GenericEdge<T> = {
  /** Item from a GraphQL connection object */
  node: T;
  /** Potential typing for the edge added by GraphQL */
  __typename: string;
};

/** Describes the generic structure of a GraphQL connection object */
type GenericConnection<T> = {
  edges: GenericEdge<T>[];
};

/**
 * Returns an array of all nodes contained in the provided GraphQL connection
 * object
 *
 * @param connection GraphQL connection object containing the nodes to return
 */
export function listNodesFromConnection<T>(
  connection?: GenericConnection<T>,
): T[] | undefined {
  return connection?.edges.map(({ node }) => node);
}
