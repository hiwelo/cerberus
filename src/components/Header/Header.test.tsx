import React from 'react';
import { testA11y, testSnapshot } from 'utilities/test';
import { Header } from './Header';

const TestContainer = <Header>Test</Header>;

describe('Header', () => {
  testSnapshot(TestContainer);
  testA11y(TestContainer);
});
