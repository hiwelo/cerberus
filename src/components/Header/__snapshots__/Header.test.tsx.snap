// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
.c0 {
  margin-bottom: 2.4rem;
  padding: 0 1.6rem;
}

.c1 {
  font-size: 2rem;
  font-weight: 600;
  -webkit-letter-spacing: -0.035em;
  -moz-letter-spacing: -0.035em;
  -ms-letter-spacing: -0.035em;
  letter-spacing: -0.035em;
  text-align: left;
}

<header
  class="c0"
>
  <h1
    class="c1"
  >
    Test
  </h1>
</header>
`;
